### PR TITLE
Adding data protection using keyvault for cookie protection in container

### PR DESCRIPTION
**Pull request by marcschier**
* Add additional monitoring command to cli
* Add data protection to application to have consistent encryption key for cookie protection.
* Deploy versioned modules instead of latest - version correlates to service version (assembly)
* Update swagger nuget packages to release version
* Fix crash when read json value was null (engineering tool)
* Add http tunnelling implementation and unit tests

### CODE REVIEW

File: common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                configuration = builder.Services.BuildServiceProvider()
+                    .GetRequiredService<IConfiguration>();
+            }
+            var config = new DataProtectionConfig(configuration);
+            if (string.IsNullOrEmpty(config.KeyVaultBaseUrl)) {
+                return builder;
+            }
+            var keyName = "dataprotection";
+            var client = TryKeyVaultClientAsync(config.KeyVaultBaseUrl,
+                config, keyName).Result;
```
Comment (mregen) [inline common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs]: Why can await not be used here?

File: common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                    vaultUri, keyName, $"RunAs=App; AppId={client.AppId}; " +
+                        $"AppKey={client.AppSecret}; TenantId={client.TenantId}");
+                if (keyVault != null) {
+                    return keyVault;
+                }
+            }
+
+            // Try using aims
+            keyVault = TryInititalizeKeyAsync("Managed Service Identity",
+                vaultUri, keyName).Result;
```
Comment (mregen) [inline common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs]: again why not await?

File: common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+            /// <summary>Application id</summary>
+            public string AppId => GetStringOrDefault("PCS_KEYVAULT_APPID",
+                Environment.GetEnvironmentVariable("PCS_KEYVAULT_APPID"))?.Trim();
+            /// <summary>App secret</summary>
+            public string AppSecret => GetStringOrDefault("PCS_KEYVAULT_SECRET",
+                Environment.GetEnvironmentVariable("PCS_KEYVAULT_SECRET"))?.Trim();
+            /// <summary>Optional tenant</summary>
+            public string TenantId => GetStringOrDefault("PCS_AUTH_TENANT",
+                Environment.GetEnvironmentVariable("PCS_AUTH_TENANT") ?? "common").Trim();
```
Comment (mregen) [inline common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs]: string dupes, better make this a constant somewhere.

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor [hunk 4]
```diff
@@ -308,7 +308,7 @@
         {
             if (node.Id == samples.NodeId)
             {
-                node.Value = samples.Value.ToObject<string>();
```
Comment (mregen) [inline samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: are these changes in razor related to PR title?

File: common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs (hunk 1 above)
Comment (marcschier) [reply common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs]: Yes, that would be nice, but convention adding services to DI is synchronous call.

File: common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs (hunk 2 above)
Comment (marcschier) [reply common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs]: Great catch.   This should be async and only the extension shoud call it sync.

File: samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor (hunk 4 above)
Comment (marcschier) [reply samples/src/Microsoft.Azure.IIoT.App/src/Pages/Browser.razor]: Good catch - this was a bug fix - it was excepting for me if value is null

File: common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+        }
+
+        /// <summary>
+        /// Dataprotection default configuration
+        /// </summary>
+        internal sealed class DataProtectionConfig : ConfigBase, IClientConfig,
+            IKeyVaultConfig, IStorageConfig {
+
+            /// <summary>Application id</summary>
+            public string AppId => GetStringOrDefault("PCS_KEYVAULT_APPID",
```
Comment (karok2m) [inline common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs]: All three new environment variables (`PCS_KEYVAULT_APPID`, `PCS_KEYVAULT_SECRET` and `PCS_AUTH_TENANT`) should be documented.

Comment (marcschier) [reply common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs]: I added a static class with all currently used PCS_* variables.   That would be the source of the documentation.  I put a work item on the backlog.

File: common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs (hunk 3 above)
Comment (marcschier) [reply common/src/Microsoft.Azure.IIoT.AspNetCore/src/Auth/Extensions/DataProtectionBuilderEx.cs]: Done

**Pull request merged**